rust/main/chains/hyperlane-starknet/src/provider.rs

```diff
@@ -0,0 +1,156 @@
+        Box::new(self.clone())
+    }
+}
+
+#[async_trait]
+impl HyperlaneProvider for StarknetProvider {
+    #[instrument(err, skip(self))]
+    async fn get_block_by_height(&self, height: u64) -> ChainResult<BlockInfo> {
+        let block = self
+            .rpc_client()
+            .get_block_with_tx_hashes(BlockId::Number(height))
+            .await
+            .map_err(Into::<HyperlaneStarknetError>::into)?;
+        match block {
+            MaybePendingBlockWithTxHashes::Block(b) => Ok(BlockInfo {
+                hash: H256::from_slice(b.block_hash.to_bytes_be().as_slice()),
+                timestamp: b.timestamp,
+                number: b.block_number,
+            }),
+            _ => Err(HyperlaneStarknetError::InvalidBlock.into()),
+        }
+    }
+
+    #[instrument(err, skip(self))]
+    async fn get_txn_by_hash(&self, hash: &H512) -> ChainResult<TxnInfo> {
+        let tx = self
+            .rpc_client()
+            .get_transaction_by_hash(
+                FieldElement::from_byte_slice_be(hash.as_bytes())
+                    .map_err(Into::<HyperlaneStarknetError>::into)?,
+            )
+            .await
+            .map_err(Into::<HyperlaneStarknetError>::into)?;
+
+        let receipt = self
+            .rpc_client()
+            .get_transaction_receipt(tx.transaction_hash())
+            .await
+            .map_err(Into::<HyperlaneStarknetError>::into)?;
+
+        match receipt {
+            MaybePendingTransactionReceipt::Receipt(tx_receipt) => match tx_receipt {
+                TransactionReceipt::Invoke(invoke_receipt) => Ok(TxnInfo {
+                    hash: H512::from_slice(tx.transaction_hash().to_bytes_be().as_slice()),
+                    gas_limit: U256::one(),
+                    max_priority_fee_per_gas: None,
+                    max_fee_per_gas: None,
+                    gas_price: None,
+                    nonce: 0,
+                    sender: H256::zero(),
+                    recipient: None,
+                    raw_input_data: None,
+                    receipt: Some(TxnReceiptInfo {
+                        gas_used: U256::from_big_endian(
+                            invoke_receipt.actual_fee.amount.to_bytes_be().as_slice(),
+                        ),
+                        cumulative_gas_used: U256::zero(),
+                        effective_gas_price: None,
+                    }),
+                }),
+                _ => Err(HyperlaneStarknetError::InvalidBlock.into()),
+            },
+            _ => Err(HyperlaneStarknetError::InvalidBlock.into()),
+        }
+    }
+
+    #[instrument(err, skip(self))]
+    async fn is_contract(&self, address: &H256) -> ChainResult<bool> {
+        Ok(true)
+    }
+
+    #[instrument(err, skip(self))]
+    async fn get_balance(&self, address: String) -> ChainResult<U256> {
+        let eth_token_address =
+            felt!("0x049d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7");
+
```

Comment:
I wonder why this address is hardcoded?

Reply:
this is the ETH token address, it's always the same will move it as a constant